make/autoconf/flags-cflags.m4

```diff
@@ -259,7 +261,9 @@ AC_DEFUN([FLAGS_SETUP_WARNINGS],
           -Wunused-function -Wundef -Wunused-value -Woverloaded-virtual"
       WARNINGS_ENABLE_ALL="-Wall -Wextra -Wformat=2 $WARNINGS_ENABLE_ADDITIONAL"
 
-      DISABLED_WARNINGS="unknown-warning-option unused-parameter unused"
+      # These warnings will never be turned on, since they generate too many
+      # false positives.
+      DISABLED_WARNINGS="unknown-warning-option unused-parameter"
```

Comment:
JDK-8339120 added -Wunused-const-variable=1 and -Wunused-result to gcc's
WARNINGS_ENABLE_ADDITIONAL, to "make up" for the corresponding removal of
disabling -Wunused. That isn't done here. I assume that was intentional,
probably to avoid needing more of the specific disables in this PR. That's
fine. We can do more warning option cleanups later in conjunction with
corresponding code changes.

Reply:
No, it was just not needed by clang. `-Wunused-result` is enabled by `-Wunused-value` which is enabled by `-Wunused` which is enabled by `-Wall`. Or something like that. I think it made sense, but the documentation was bad so you had to jump around to understand how things were related.

`Wunused-const-variable=1` does not exist in clang, but whatever the closest matching thing was, was enabled as well indirectly already.